### PR TITLE
Add slog utils

### PR DESCRIPTION
**Pull request by heppu**

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"io"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+)
+
+const DefaultRefreshInterval = time.Second * 5
+
+// NewDefaultEnvLogger creates new slog.Logger using sane default configuration and sets it as a default logger.
```
Comment (kraashen) [inline log/logger.go:14]: Documented function name `NewDefaultEnvLogger` does not match the declared `NewDefaultLogger`. Should rename to `.*Env.*` as it seems to get the log level from environment or adjust the documented function name? What should `NewDefaultLogger` do?

File: log/logger.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	opts := &slog.HandlerOptions{
+		AddSource: true,
+		Level:     lvl,
+	}
+
+	logger := slog.New(handlerFn(os.Stdout, opts))
+	slog.SetDefault(logger)
+
+	return logger
+}
```
Comment (kraashen) [inline log/logger.go:37]: Would we like to also have a `NewLogger` utility handler with which we could set the desired log level as a function argument, or leave the implementation of a new logger with custom log level to the user? Right now it is fetched from the environment which can be later set with `RefreshLogLevel`. Should we allow users also set log level programmatically on logger initialization?

Comment (heppu) [reply log/logger.go:37]: I feel like there's slog already gives the nice building blocks for that so there should be no need for that. Also I don't see usecase for setting log level manually or atleast haven't spotted it anywhere in our code base yet.

File: log/logger.go (hunk 1 above)
Comment (heppu) [reply log/logger.go:14]: Fixed

**Review by kraashen — APPROVED** [review]
LGTM +1

**Pull request merged**